message/retrylater.go

```diff
@@ -22,19 +22,20 @@ type retryLaterHandler struct {
 }
 
 func (r *retryLaterHandler) Do(ctx context.Context, orig Handler, message *servicebus.Message) Handler {
-	go func() {
```

Comment:
no need to have this in a go routine. each msg handler is already processed in a goroutine. let it block and make it simpler